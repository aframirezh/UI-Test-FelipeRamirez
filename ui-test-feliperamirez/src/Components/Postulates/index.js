import React, { Component } from 'react';
import '../../Containers/Card/styles.scss';
import VoteButton from '../VoteButton';

class Postulates extends Component {
    constructor(props) {
        super(props);
        this.state = {
        }
    }

    render() {
        const info = this.props;

        var head = (
            <div>
                <section>
                    <p>
                        <small className="thin">What's your opinion on</small>
                        <br />
                        <strong className="box-tittle-size regular">{info.name}</strong>
                    </p>
                    <p className="box-content-size light test-head-des">{info.description}</p>
                    <p>
                        <img src="https://upload.wikimedia.org/wikipedia/commons/5/5a/Wikipedia%27s_W.svg"
                            alt="wikipedia" className="wiki" />
                        <a href="#wiki">
                            More information
                            </a>
                    </p>
                    <p className="box-content-size regular">What's Your Verdict?</p>
                </section>
            </div>
        );

        var card = (
            <section>
                <p>
                    <strong className="box-tittle-votes-size regular">{info.name}</strong>
                </p>
                <p className="thin">
                    {info.timeHere}
                </p>
                <p className="regular">{info.description}</p>
                <VoteButton id={info.id}/>
            </section>
        );

        return (
            <div>
                {info.cardFlag ? head : card}
            </div>
        );
    }
}

export default Postulates;