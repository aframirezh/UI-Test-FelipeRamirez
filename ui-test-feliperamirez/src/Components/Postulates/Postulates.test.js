import React from 'react'
import Postulates from '.'

import renderer from 'react-test-renderer';

it ('Postulate render correctly', ()=>{
        const props = {
            cardFlag: "head",
            name: 'Test Name',
            description: 'Description test',
            id: 1
        }
        const component = renderer.create(
            (<Postulates {...props}/>),
        );
        let postulate = component.toJSON();
        const nameNode = component.root.findByType("strong");
        expect(nameNode.children[0]).toBe(props.name);
    
        const srcImage = component.root.findByType("img");
        expect(srcImage.props.className.includes("wiki")).toBe(true);
        // [ 3 ] query for element
        const descriptionNode = component.root.findAllByType("p")[1];
        expect(descriptionNode.props.className.includes("test-head-des")).toBe(true);
        expect(postulate).toMatchSnapshot();

})