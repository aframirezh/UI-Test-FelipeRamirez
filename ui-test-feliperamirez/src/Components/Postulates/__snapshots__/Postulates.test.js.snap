// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Postulate render correctly 1`] = `
<div>
  <div>
    <section>
      <p>
        <small
          className="thin"
        >
          What's your opinion on
        </small>
        <br />
        <strong
          className="box-tittle-size regular"
        >
          Test Name
        </strong>
      </p>
      <p
        className="box-content-size light test-head-des"
      >
        Description test
      </p>
      <p>
        <img
          alt="wikipedia"
          className="wiki"
          src="https://upload.wikimedia.org/wikipedia/commons/5/5a/Wikipedia%27s_W.svg"
        />
        <a
          href="#wiki"
        >
          More information
        </a>
      </p>
      <p
        className="box-content-size regular"
      >
        What's Your Verdict?
      </p>
    </section>
  </div>
</div>
`;
